apis/XCast/IXCast.h

```diff
@@ -26,51 +26,199 @@
+			virtual Core::hresult GetEnabled(bool &enabled /* @out @text enabled */, bool &success /* @out */) = 0;
+			/***************************************** getEnabled() **********************************/
+
+			/****************************************setStandbyBehavior()**********************************/
+			// @text setStandbyBehavior
+			// @brief Sets the expected xcast behavior in standby mode
+			// @param standbybehavior: whether to remain active or inactive during standby mode (must be one of the following: active, inactive)
+			virtual Core::hresult SetStandbyBehavior(const StandbyBehavior &standbybehavior /* @in @text standbybehavior */, bool& success /* @out */) = 0;
+			/***************************************** setStandbyBehavior() *********************************/
+
+			/****************************************getStandbyBehavior()**********************************/
+			// @text getStandbyBehavior
+			// @brief Return current standby behavior option string set uisng setStandbyBehavior or default value
+			// @param standbybehavior: whether to remain active or inactive during standby mode (must be one of the following: active, inactive)
+			// @param success: Whether the request succeeded
+			virtual Core::hresult GetStandbyBehavior(StandbyBehavior &standbybehavior /* @out @text standbybehavior */, bool &success /* @out */) = 0;
+			/***************************************** getStandbyBehavior() *********************************/
+
+			/****************************************setFriendlyName()**********************************/
+			// @text setFriendlyName
+			// @brief Sets the friendly name of the device
+			// @param friendlyname: The friendly name of the device which used to display on the client device list
+			virtual Core::hresult SetFriendlyName(const string &friendlyname /* @in @text friendlyname */, bool& success /* @out */) = 0;
+			/***************************************** setFriendlyName() **********************************/
+
+			/****************************************getFriendlyName()**********************************/
+			// @text getFriendlyName
+			// @brief Returns the friendly name set by setFriendlyName API
+			// @param friendlyname: The friendly name of the device which used to display on the client device list
+			// @param success: Whether the request succeeded
+			virtual Core::hresult GetFriendlyName(string &friendlyname /* @out @text friendlyname */, bool &success /* @out */) = 0;
+			/***************************************** getFriendlyName() *********************************/
+
+			/****************************************getApiVersionNumber()**********************************/
+			// @text getApiVersionNumber
+			// @brief Gets the API version number
+			// @param version: a version number
+			// @param success: Whether the request succeeded
+			virtual Core::hresult GetApiVersionNumber(uint32_t &version /* @out @text version */, bool &success/* @out */) = 0;
```

Comment:
Pls remove this API. this is legacy requirement which is not required any more. we can use Controller.Status API to query plugin version